File: week02/Theme03/Theme03_Code01.cpp
// Theme03_Code01.cpp

#include <iostream>
#include <vector>
using namespace std;

int main() {
	ios_base::sync_with_stdio(false);
	cin.tie(NULL);
	cout.tie(NULL);

	int n, m;
	cin >> n >> m;
	vector<vector<bool>> avoid(n + 1, vector<bool>(n + 1, false));
	for (int i = 0; i < m; i += 1) {
		int a, b;
		cin >> a >> b;
		avoid[a][b] = true;
		avoid[b][a] = true;
	}

	int res = 0;
	for (int i = 1; i <= n; i += 1) {
		for (int j = i + 1; j <= n; j += 1) {
			for (int k = j + 1; k <= n; k += 1) {
				if (!avoid[i][j] && !avoid[j][k] && !avoid[k][i]) res += 1;
			}
		}
	}

	cout << res << endl;
    return 0;
}

File: week02/Theme03/Theme03_Code01.py
# Theme03_Code01.py

n, m = map(int, input().split())
avoid = [[False] * (n + 1) for _ in range(n + 1)]

for _ in range(m):
    a, b = map(int, input().split())
    avoid[a][b] = True
    avoid[b][a] = True

res = 0
for i in range(1, n + 1):
    for j in range(i + 1, n + 1):
        for k in range(j + 1, n + 1):
            if not (avoid[i][j] or avoid[j][k] or avoid[k][i]):
                res += 1

print(res)